js/packages/components/settings/NetworkMap.js

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { ScrollView, View, Text as TextNative, ActivityIndicator } from 'react-native'
+import { Layout, Text, Icon } from '@ui-kitten/components'
+import { HeaderSettings } from '../shared-components/Header'
+import { useStyles } from '@berty-tech/styles'
+import { useNavigation } from '@berty-tech/navigation'
+import { protocolMethodsHooks } from '@berty-tech/store/methods'
+import { SwipeNavRecognizer } from '../shared-components/SwipeNavRecognizer'
+
+const PeerItem = ({ id, latency, highlighted }) => {
+	const [{ padding, border, color, text, flex, row }] = useStyles()
+	return (
+		<View
+			style={[
+				padding.small,
+				border.scale(1),
+				border.color.light.grey,
+				border.radius.small,
+				{ justifyContent: 'space-evenly', flexDirection: 'row', alignItems: 'center' },
+				highlighted && { backgroundColor: color.light.yellow },
+			]}
+		>
+			<View style={[flex.tiny, row.center]}>
+				<Icon name='earth' pack='custom' fill={color.dark.grey} width={25} height={25} />
+			</View>
+			<View style={[flex.tiny, row.center]}>
+				<Icon name='network' pack='custom' fill={color.dark.grey} width={25} height={25} />
+			</View>
+			<Text style={[flex.medium, text.align.center]}>{id.substr(0, 9)}</Text>
+			<Text style={[flex.tiny, text.align.center]}>{latency ? latency + 'ms' : '?'}</Text>
+			<View style={[flex.tiny, row.center]}>
+				<Icon name='arrow-ios-downward' fill={color.dark.grey} width={25} height={25} />
+			</View>
+		</View>
+	)
+}
+
+function usePrevious(value) {
+	// https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+	const ref = useRef()
+	useEffect(() => {
+		ref.current = value
+	})
+	return ref.current
+}
+
+const NetworkMapBody = ({ refresh, setRefresh }) => {
+	const { reply: peers = {}, call, called } = protocolMethodsHooks.usePeerList()
+	const [{ margin, text }] = useStyles()
+	const [sortPeers, setSortPeers] = useState(null)
+
+	const prevPeers = usePrevious(sortPeers)
+
+	useEffect(() => {
+		if (!called || refresh) {
+			if (refresh) {
+				setRefresh(false)
+			}
+			call()
+		}
+	}, [called, call, refresh, setRefresh])
```

Comment:
you can avoid this whole block by moving the method hook in the parent component and passing `call` as a prop directly

Reply:
yes it's part the part of code that i'm not proud but i didn't know how to do that properly but okay 👍